uyuni-tools/last_update_export.py

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/python3
+"""
+Script Name: Last Update Export
+
+Description:
+This script automates the export of software channels using an XML-RPC client to interface with a server for use in Airgapped Environments or Disconnected SUMA.
+It is designed to operate on a scheduled basis (daily), removing the previous day's export data, and writing new export logs into designated directories.
+The concept behind the script is to use the API's to determine which channels have been updated within the defined dates, and export only those channels, as such, 
+this creates a more streamlined export/import for Disconnected or Airgapped systems.
+
+Instructions:
+1. Ensure Python 3.x is installed on your system.
+2. This script must be run with root privileges to manage file permissions and perform system-level operations.
+3. Before running the script, update the `/root/.mgr-sync` configuration file with the correct manager login credentials using `mgr-sync -s refresh` to create the credentials file.
+4. Customize the directory path, by default `/mnt` was chosen but this could be any location you want, ensure the location has ample free space.
+5. Customize the 'rsync_user' and 'rsync_group' in the script to match the user and group names on your system.
+6. Schedule this script using a cron job or another scheduler for daily execution.
+
+Intended Usage:
+- Scheduled daily exports of software channel data.
+- Logging of export operations in '/mnt/logs'.
+- Exports are stored in '/mnt/export/updates'.
+- This script is intended for systems administrators managing software channel updates for Airgapped Environments.
+
+Ensure the server and paths are correctly configured and accessible before running the script.
+"""
+
+import os
+import subprocess
+import datetime
+from xmlrpc.client import ServerProxy
+import ssl
+import socket
+import configparser
+import shutil
+
+def setup_directories(base_path, output_path, log_path):
+    if not os.path.exists(base_path):
+        os.makedirs(base_path, exist_ok=True)
+    if not os.path.exists(log_path):
+        os.makedirs(log_path, exist_ok=True)
+    if os.path.exists(output_path):
+        shutil.rmtree(output_path)
+    os.makedirs(output_path, exist_ok=True)
+
+def setup_logging(log_dir, today):
+    log_file_path = os.path.join(log_dir, f"{today}-daily_export.log")
+    return log_file_path
+
+def create_client():
+    config_path = os.path.expanduser('/root/.mgr-sync')
+    config = configparser.ConfigParser()
+    with open(config_path, 'r') as f:
+        config.read_string('[DEFAULT]\n' + f.read())
+    MANAGER_LOGIN = config.get('DEFAULT', 'mgrsync.user')
+    MANAGER_PASSWORD = config.get('DEFAULT', 'mgrsync.password')
+    SUMA_FQDN = socket.getfqdn()
+    MANAGER_URL = f"https://{SUMA_FQDN}/rpc/api"
+    context = ssl.create_default_context()
+    client = ServerProxy(MANAGER_URL, context=context)
+    return client, client.auth.login(MANAGER_LOGIN, MANAGER_PASSWORD)
```

Comment:
I would suggest to not use the uppercase notation here, as these values are not hardcoded anymore, and contants are usually defined at the top level of the file.

Maybe just go with something like:
```suggestion
    manager_login = config.get('DEFAULT', 'mgrsync.user')
    manager_password = config.get('DEFAULT', 'mgrsync.password')
    suma_fqdn = socket.getfqdn()
    manager_url = f"https://{suma_fqdn}/rpc/api"
    context = ssl.create_default_context()
    client = ServerProxy(manager_url, context=context)
    return client, client.auth.login(manager_login, manager_password)
```

Reply:
modified.